modules/ting_subsearch/ting_subsearch.module

```diff
@@ -0,0 +1,239 @@
+<?php
+
+/**
+ * @file
+ * Ting Subsearch
+ */
+
+use Ting\Search\TingSearchResultInterface;
+
+/**
+ * Implements hook_menu().
+ */
+function ting_subsearch_menu() {
+  $items = [];
+
+  $items['admin/config/ding/subsearch'] = [
+    'title' => 'Ting subsearch',
+    'description' => 'Configure ting subsearch triggers and settings.',
+    'page callback' => 'drupal_get_form',
+    'page arguments' => array('ting_subsearch_admin_settings_form'),
+    'access arguments' => array('administer site configuration'),
+  ];
+
+  return $items;
+}
+
+/**
+ * Admin settings form for subsearch modules.
+ */
+function ting_subsearch_admin_settings_form($form, &$form_state) {
+  $form = [];
+
+  $form['subsearch'] = [
+    '#type' => 'fieldset',
+    '#title' => t('Ting Subsearch'),
+    '#description' => t('Subsearch modules allow additional searches to be made based on predefined conditions.'),
+  ];
+
+  return system_settings_form($form);
```

Comment:
But then we would have an empty page with three local tabs. There are no general settings. Perhaps that is fine?

But tbh I don't really see this as a big issue. All modules add their settings under their own module key. There are only three modules.

Reply:
It works better with ting_covers since it has a general settings form.

Maybe I can add some general documentation to the default page. I'll look into it.

Reply:
Added new `hook_ting_subsearch_info` hook where subsearch modules return info about their settings. Ting subsearch now sets up the form based on the info returned by this hook, which should give more control.

Removed form alter hook from subsearch modules.

Reply:
I completely removed any admin settings form handling from the main ting_subsearch module. Now each module add their own tab with settings under admin/config/ding/subsearch. Ting subsearch suggestions could be consideres the general "default" subsearch module and is therefore providing the default tab.